src/components/step/ConnectDeviceAccountStepContainer/ConnectDeviceAccountStepContainer.tsx

```diff
@@ -36,7 +36,7 @@ export default function () {
 
     useEffect(() => {
         // Start polling for connected status after a providerID is selected.
-        if (!providerID || connected) return;
+        if (!providerID || connected || !MyDataHelps.token) return;
```

Comment:
Not sure I understand why this is needed. Doesn't getExternalAccounts() set the token?

Reply:
But yeah, as @ashleyrtate pointed out, I think the token actually would get set by the first call to getExternalAccounts() (because that is the first API request).

Actually, backing up, what happens if they visit this step and they've already connected Fitbit?  (I can't test it in QA right now since it seems to be complaining about missing providerID)

Reply:
We probably want to check if they have already connected the device and auto-proceed if so; so I think when this step loads you would need to call getExternalAccounts immediately...

Reply:
1. A _MyDataHelpsDesigner_ user add a ConnectDeviceAccountStep to a survey.
2. They set a providerID.
3. At this time, the rendered ConnectDeviceAccountStep step starts polling, which always fail due to lack of token.

This PR aims to address this issue.

Reply:
I would set a "designMode" flag or something to make that explicit.

Reply:
@chrisnowak @ashleyrtate ^

Reply:
I thought it should only start polling after the button is clicked?

Reply:
No, this is the ConnectDeviceAccountStep.

> When the user visits this step it would check to see if they have the account connected; if so it would automatically proceed.

https://github.com/CareEvolution/Consumers/issues/11603

Reply:
Ah sorry I didn't notice this in the original PR - there was another issue I also was wondering about the other day:
- Seems like we should only check once when the step is loaded vs. always poll, right?
- Also...curious which / what platform is forcing us to poll anyways?  ConnectExternalAccount should return a promise so you should know when it is complete, no?

Reply:
Polling is needed as we wait for the participant to connect to the device account. But you are right that this process can start **after** the participant clicks the button. In the initialization phase, we can just make one request.

Reply:
(I'm working on the change)

Reply:
@bxh why is it needed?  We get a promise back when they click the button that resolves when they connect the account

Reply:
Clicking the button only invokes `.connectExternalAccount`. That does not mean the participant is instantaneously connected, right? They still need to proceed to the third party website to sign in themselves.

Reply:
Right...after they do the connection the promise returned by connectExternalAccount resolves with a status (or fails if they cancel I think)

Reply:
e.g. you should be able to do something like
```
MyDataHelps.connectExternalAccount(1).then(() => {
   //stuff that should happen after connection is complete
});
```

Reply:
Oh, I didn't realize that `connectExternalAccount` resolves _after_ a connection is complete.